lib/linguist/languages.yml

```
@@ -4527,6 +4527,16 @@ Nix:
   tm_scope: source.nix
   ace_mode: nix
   language_id: 252
+Noir:
+  type: programming
+  aliases:
+  - nargo
+  ace_mode: text
```

Rust's [CodeMirror](https://codemirror.net/5/mode/rust/index.html) and [ACE](https://ace.c9.io/build/kitchen-sink.html) editor modes do a good job of highlighting Noir source:

```suggestion
  ace_mode: rust
  codemirror_mode: rust
  codemirror_mime_type: text/x-rustsrc
```